system/Validation/FormatRules.php

```diff
@@ -334,11 +334,14 @@ public function valid_url(string $str = null): bool
 		}
 		elseif (preg_match('/^(?:([^:]*)\:)?\/\/(.+)$/', $str, $matches))
 		{
-			if (empty($matches[2]))
-			{
-				return false;
-			}
-			elseif ( ! in_array($matches[1], ['http', 'https'], true))
+// removed the test for an empty second match segment. 
+// it should never be found, because of the "(.+)" pattern, which is one or more
```

Comment:
Might as well remove the code if it's not needed. you can leave the comment so we don't try to put it back in down the road, though.

Reply:
Done, but I removed the comment since it looked out of place without the commented code.